### PR TITLE
feat(LUKE-152): fold the latest rating into the view's message

### PR DESCRIPTION
**Pull request by dastratakos**
## What

The view's message now carries the developer's latest rating, folded from the rating events exactly as an announcement's `unspoken` mark is folded from the speech events.

- **View (`@sidecar/session`):** `ConversationViewEvent` gains an optional `rating` (the row's payload read under `RATING_EVENT_PAYLOAD`), and `ConversationViewMessage` gains an optional `rating`: the newest rating event on the message by its conversation's event sequence. A newest rating whose payload could not be read folds as no rating rather than an older verdict, matching C6's `latestMessageRating`.
- **Route:** the events the page already reads for the speech fold now carry their rating payload into the view, and the answer's message carries the folded `rating`.
- **Wire (`@sidecar/hosted`):** `ConversationReadMessage.rating?` is `RATING_EVENT_PAYLOAD.optional()`, the vocabulary's own schema declared once. The shared fixture carries a rating on the observed announcement.
- **The record is unchanged.** `events` stays append-only, one row per rating, a re-rating superseding by being newer. This changes the projection, which is what the view is for.

## Why

The messages answer folded a briefing's speech marks but not its rating, so every client that wanted to show an earlier rating had to read the `events` resource from its beginning, paged, once per launch, and that read grows without bound. Per-resource cursors exist so a client moves forward from where it was; requiring a full read of one resource to render the current state of another is the shape they were meant to avoid.

## Tests

- **View:** no rating without a rating event; the latest by sequence wins whatever order the events arrive in; a re-rating replaces the fold; an unreadable newest rating folds as none; a rating on another message does not attach.
- **Route, over PGlite:** a device that has never read the events resource sees the correct rating on its first messages page; a re-rating changes what the next read answers; and **the events record still holds both rating rows in order, with their payloads**, both through the table and through the store's events read. That last assertion is the one that guards this PR against a later "simplification" of the record behind the fold.
- **Wire:** the fixture's rated message reads back with its rating and an unrated one without.

## Client half (LukeKit), on F4

F4 (#990) merged while the server half was under check, so the client pass is in this PR.

- `ConversationReadMessage` gains `rating: RatingEventPayload?`, a new Swift struct mirroring the wire's payload (verdict and optional note), decoded into F4's `MessageRating`; the shared fixture's rated announcement decodes to it.
- `ConversationThread.ratings` is now the rating the service folded onto each message, **amended** by the newer rating events read since, including the one this device just wrote. The events sweep stays: later ratings and speech marks on messages the cursor has passed still arrive only as events. What goes is the read **from the beginning**: `adoptHeads` now seeds the events cursor from the change signal's head exactly as it seeds the turns, so a fresh launch reads no events it does not need.
- **The `eventsCaughtUp` gate stays, and is not removed.** It guarded marks replayed from the record's beginning, which are older than the fold. That replay still happens on one path (a thread that first read messages without a device id, then gained one), and it is the right gate for a first head-seeded read that pages. Seeding from the head sets the gate, since there is nothing older than the fold to replay, which is what lets this device's own rating show at once.
- `ConversationStore`'s `ratingClient` is optional, so a screen that draws no control constructs no client it never calls; `canRate` requires one. The phone passes its client as before; F5's watch can omit it.

**Swift verified on Linux.** With a Swift 6.2 toolchain and a scratch package over LukeKit's sources (the path F1, F4, and F5 used), 259 tests pass against the shared fixture bytes, `ConversationThreadTests`, `ConversationReadsFixtureTests`, and `ConversationTurnRowsTests` among them. Set aside on Linux only: the four Apple-framework sources (`KeychainStore` stood in by an in-memory copy, `Marks`, `PKCE`, `PCMAudio`) and ten test files that either need those frameworks or use `@MainActor` test methods Linux's XCTest cannot dispatch; none of them touches this change. The run caught one logic error re-reading had not: a rating this device had just written was hidden behind the events gate, which exists only for replayed events older than the fold. `record` now marks its event as this device's own, which amends at once, while events read back still wait for the gate; a test pins both sides. CI builds no LukeKit target, which the orchestrator is raising separately.

## CLAUDE.md

No sentence contradicted. `apps/web/server/README.md` moves with the behaviour.

## Reviews

- **Cursor thermo-nuclear code quality review**: applied as written. The fold is one more derived map beside `latestSpeechEvents`, threaded through `viewMessage` as a parameter rather than a closure over module state; no new branch in the walk or the route.
- **Ponytail review**: applied as written. One filler assertion removed from the new route test; F4's per-event rating fold is replaced by reading the field rather than kept beside it; the optional `DeviceTouch` comment the brief offered was skipped to keep the diff to the fix.

## Verify

```sh
./scripts/check.sh
```

Passes locally. No macOS or UI code is touched, so `verify.sh` does not apply.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### Automated visual evidence

[Download the deterministic macOS evidence](https://github.com/ReviewStage/luke/actions/runs/34565708113/artifacts/10185973508) · [workflow run](https://github.com/ReviewStage/luke/actions/runs/34565708113)

- Commit: `72f4e39967c5cfd1542f90043658ecfa589fa788`
- Scenario: `smoke`
- Physical-notch check: not performed by CI

### CODE REVIEW

File: apps/ios/LukeKit/Sources/LukeKit/ConversationThread.swift [hunk 1]
```diff
@@ -120,24 +121,36 @@ public struct ConversationThread: Equatable, Sendable {
...
-        latestRating.mapValues(\.rating)
+        var verdicts: [String: MessageRating] = [:]
+        for group in groups.values {
+            for message in group.messages.values {
+                if let rating = message.rating?.rating { verdicts[message.message.id] = rating }
+            }
+        }
+        guard eventsCaughtUp else { return verdicts }
+        for (id, mark) in latestRating { verdicts[id] = mark.rating }
+        return verdicts
```
[the remark this answers is not in the document]
Comment (dastratakos) [reply apps/ios/LukeKit/Sources/LukeKit/ConversationThread.swift:148]: Fixed in b8e096a0, found the same way by running LukeKit's tests on Linux before enqueueing: record() now marks the event as this device's own, and an own mark amends the fold at once, while events read back still wait for eventsCaughtUp so a partial replay cannot override the service's fold. ConversationThreadTests pins both sides (an unread thread's own rating shows immediately; a replayed rating waits until the read has caught up).
